test/valid-data/enums-union/main.ts

```diff
@@ -0,0 +1,16 @@
+enum Alphabet {
+    Alpha = "alpha",
+    Beta = "beta",
+    Omega = 666,
+}
+
+export type MyObject = {
+    // All the members above should be output as enums, not anyOf
+    enumMembers: Alphabet.Alpha | Alphabet.Beta;
+    enumMemberWithLiteral: Alphabet.Alpha | "foo";
+    enumMemberWithLiteralAndNull: Alphabet.Alpha | "foo" | null;
+    enumMembersWithNumber: Alphabet.Alpha | Alphabet.Omega;
+    wholeEnum: Alphabet; // Should output just all of Alphabet
+    wholeEnumWithLiteral: Alphabet | "bar"; // Should output all of Alphabet members (2 strings, 1 number) and "bar"
+    wholeEnumWithLiteralAndNull: Alphabet | "bar" | null;
```

Comment:
could you add a test with a union of two enums? e.g: `Alphabet | Numerals`

Reply:
please also add a test with a literal/enum + something non literal to ensure its still `anyOf`

Reply:
Sure!